### PR TITLE
[Model] Pipeline parallel support for Mixtral

### PR DESCRIPTION
**Pull request by comaniac**
Take from #6403. Co-authored by @binxuan

cc @youkaichao

### CODE REVIEW

**Comment by comaniac** [comment]
Tested locally with PP=8 and worked.

**Comment by youkaichao** [comment]
can you test the correctness locally, using https://github.com/vllm-project/vllm/blob/main/tests/distributed/test_pipeline_parallel.py ?

**Comment by comaniac** [comment]
Passed with the following configures. Note that I tested it on 8xL4 so I have to use 8 GPUs to host the model.

```
    "TP_SIZE, PP_SIZE, EAGER_MODE, CHUNKED_PREFILL, MODEL_NAME",
    [
        (2, 4, 0, 1, "mistralai/Mixtral-8x7B-Instruct-v0.1"),
        (2, 4, 1, 0, "mistralai/Mixtral-8x7B-Instruct-v0.1"),
        (1, 8, 0, 1, "mistralai/Mixtral-8x7B-Instruct-v0.1"),
        (1, 8, 1, 0, "mistralai/Mixtral-8x7B-Instruct-v0.1"),
    ])
```

Also fixed some issues in the test file:
- Use TP_SIZE x PP_SIZE as the TP size of the reference. The current `max(TP_SIZE, 2)` doesn't work for larger models.
- Do not use 0's as the token ID. This may generate random outputs for certain models/tokenizers.

File: tests/distributed/test_pipeline_parallel.py [hunk 1]
```diff
@@ -34,7 +37,9 @@ def test_compare_tp(TP_SIZE, PP_SIZE, EAGER_MODE, CHUNKED_PREFILL, MODEL_NAME):
         "--dtype",
         "bfloat16",
         "--tensor-parallel-size",
-        str(max(TP_SIZE, 2)),  # use at least TP_SIZE=2 to hold the model
+        # Use the same number or at most 8 GPUs to hold the model.
+        # In this test we assume the model can fit in 8 GPUs.
+        str(min(TP_SIZE * PP_SIZE, 8)),
```
Comment (youkaichao) [inline tests/distributed/test_pipeline_parallel.py:42]: it's not going to work. this will run in multi-node tests with mp backend, and we can use at most 2 GPUs.

you can revert this change, keep it only for your local testing.

Comment (comaniac) [reply tests/distributed/test_pipeline_parallel.py]: Reverted with comments.

**Review by youkaichao — APPROVED** [review]
LGTM if tests pass
